src/components/Views/ObservationEditor/ObservationEditor.js

```diff
@@ -403,20 +404,38 @@ class ObservationEditor extends React.Component<
     Keyboard.dismiss();
   };
 
+  goToObservationFields = () => {
+    const { goToObservationFields } = this.props;
+    goToObservationFields();
+    Keyboard.dismiss();
+  };
+
   render() {
     const {
       navigation,
       goBack,
       selectedObservation,
-      goToPhotoView
+      goToPhotoView,
+      goToObservationFields
     } = this.props;
     const { keyboardShown, text } = this.state;
     const positionText = selectedObservation
       ? `${selectedObservation.lat}, ${selectedObservation.lon}`
       : 'Loading...';
     const keyExtractor = item => item.source;
+
+    let hasFieldAnswered = false;
+    if (selectedObservation) {
+      selectedObservation.fields.forEach(field => {
+        if (field.answered) hasFieldAnswered = true;
```

Comment:
what is hasFieldAnswered supposed to do? Right now it looks like it just defaults to what the last value in `selectedObservation.fields.answered` is?

Reply:
Helps with determining if we should show the grey or orange check below

Reply:
Why enumerate across fields? Are you trying to check if at least one field in fields is answered?

Reply:
Yaaaa

Reply:
Maybe use find? Just so it's more clear what's happening here? something like
`const fieldAnswered = selectedObservation.fields.find(f => f.answered)`?